src/actions/bitcoinClient.ts

```diff
@@ -80,8 +80,16 @@ export const scanTxOutSet = async (
 // Function to get a raw transaction
 export const getRawTransaction = async (
   txid: string,
-): Promise<BitcoinTransactionResponse> => {
+): Promise<BitcoinTransactionResponse | null> => {
   const result = await fetch(`${env.MEMPOOL_API_URL}/tx/${txid}`);
+  if (result.status === 404) {
+    return null;
+  }
+  return await result.json();
+};
+
+export const getTxRbf = async (txid: string): Promise<boolean> => {
+  const result = await fetch(`${env.MEMPOOL_API_URL}/v1/tx/${txid}/rbf`);
```

Comment:
Should this also have the `result.status` guard?

Reply:
actually no, and I'm being optimistic here tbh, basically the user can do rbf which could cancel the transaction which breaks everything, that case is not handled